pkg/install/1-installresources.go

```diff
@@ -88,557 +82,567 @@ func (i *Installer) installResources(ctx context.Context) error {
-						Location: &installConfig.Config.Azure.Region,
-					},
-					APIVersion: apiVersions["network"],
-					DependsOn: []string{
-						"Microsoft.Network/publicIPAddresses/aro-outbound-pip",
 					},
+					Name:     to.StringPtr("aro"),
+					Type:     to.StringPtr("Microsoft.Network/loadBalancers"),
+					Location: &installConfig.Config.Azure.Region,
+				},
+				APIVersion: apiVersions["network"],
+				DependsOn: []string{
+					"Microsoft.Network/publicIPAddresses/aro-outbound-pip",
 				},
 			},
-		}
+		},
+	}
 
-		i.log.Print("deploying resources template")
-		err = i.deployments.CreateOrUpdateAndWait(ctx, resourceGroup, "azuredeploy", mgmtresources.Deployment{
-			Properties: &mgmtresources.DeploymentProperties{
-				Template: t,
-				Parameters: map[string]interface{}{
-					"sas": map[string]interface{}{
-						"value": map[string]interface{}{
-							"signedStart":         i.doc.OpenShiftCluster.Properties.Install.Now.Format(time.RFC3339),
-							"signedExpiry":        i.doc.OpenShiftCluster.Properties.Install.Now.Add(24 * time.Hour).Format(time.RFC3339),
-							"signedPermission":    "rl",
-							"signedResourceTypes": "o",
-							"signedServices":      "b",
-							"signedProtocol":      "https",
-						},
+	i.log.Print("deploying resources template")
+	err = i.deployments.CreateOrUpdateAndWait(ctx, resourceGroup, "azuredeploy", mgmtresources.Deployment{
+		Properties: &mgmtresources.DeploymentProperties{
+			Template: t,
+			Parameters: map[string]interface{}{
+				"sas": map[string]interface{}{
+					"value": map[string]interface{}{
+						"signedStart":         i.doc.OpenShiftCluster.Properties.Install.Now.Format(time.RFC3339),
+						"signedExpiry":        i.doc.OpenShiftCluster.Properties.Install.Now.Add(24 * time.Hour).Format(time.RFC3339),
+						"signedPermission":    "rl",
+						"signedResourceTypes": "o",
+						"signedServices":      "b",
+						"signedProtocol":      "https",
 					},
 				},
-				Mode: mgmtresources.Incremental,
 			},
-		})
-		if err != nil {
-			if isDeploymentActiveError(err) {
-				i.log.Print("waiting for resource template")
-				err = i.deployments.Wait(ctx, resourceGroup, "azuredeploy")
-			}
-			isQuota, errMsg := isResourceQuotaExceededError(err)
-			if isQuota {
-				return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeQuotaExceeded, errMsg, "")
-			}
-			return err
+			Mode: mgmtresources.Incremental,
+		},
+	})
+	if err != nil {
+		if isDeploymentActiveError(err) {
+			i.log.Print("waiting for resource template")
+			err = i.deployments.Wait(ctx, resourceGroup, "azuredeploy")
+		} else if isQuota, errMsg := isResourceQuotaExceededError(err); isQuota {
+			err = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeQuotaExceeded, errMsg, "")
 		}
 	}
+	return err
+}
 
-	return nil
+// zones verifies that the Azure zone information for the control plane is correct.
+// It returns a list of zones if it is or raises an error if it is not.
```

Comment:
zones configures how master nodes are distributed across availability zones.  In regions where the number of zones matches the number of nodes, it's one node per zone.  In regions where there are no zones, it's all nodes in the same place.  Anything else (e.g. 2-zone regions) is currently unsupported.

Reply:
Thanks - I will replace my comment with this better explanation :)